lib/Runtime/Language/InterpreterLoop.inl

```diff
@@ -25,6 +25,126 @@
 #define INTERPRETERPROFILE 0
 #define PROFILEDOP(prof, unprof) unprof
 #endif
+
+//two layers of macros are necessary to get arguments to the invocation of the top level macro expanded.
```

Comment:
In general, this expansion trick is not needed. However, in case the entry level macro receives another macro as an input, single layer wouldn't work. In other words, initial macro call would expand over caller. That's the reason for multiple layers.